src/metabase/query_processor/streaming.clj

```diff
@@ -81,21 +82,23 @@
                                             m')))
                                       {}
                                       cols-vector)]
-    (->> (map
-          (fn [{field-ref ::mb.viz/table-column-field-ref, col-name ::mb.viz/table-column-name}]
-            (let [index              (or (get cols-index field-ref)
-                                         (get cols-index col-name))
-                  col                (get cols-vector index)
-                  remapped-to-name   (:remapped_to col)
-                  remapped-from-name (:remapped_from col)]
-              (cond
-                remapped-to-name
-                (get cols-index remapped-to-name)
-
-                (not remapped-from-name)
-                index)))
-          enabled-table-cols)
-         (remove nil?))))
+    (if pivot?
+      (range (count cols))
+      (->> (map
+            (fn [{field-ref ::mb.viz/table-column-field-ref, col-name ::mb.viz/table-column-name}]
+              (let [index              (or (get cols-index field-ref)
+                                           (get cols-index col-name))
+                    col                (get cols-vector index)
+                    remapped-to-name   (:remapped_to col)
+                    remapped-from-name (:remapped_from col)]
+                (cond
+                  remapped-to-name
+                  (get cols-index remapped-to-name)
+
+                  (not remapped-from-name)
+                  index)))
+            enabled-table-cols)
+           (remove nil?)))))
```

Comment:
this is kind of a big chunk to do in the if branch, consider extracting it? Also I think you can divert based on `pivot?` sooner.